doc/upstream_release_process.md

```diff
@@ -1,152 +1,166 @@
+
+## Create upstream-release branch
+**SAVE ALL LOCAL CHANGES TO A BRANCH**, then
+```bash
+git fetch upstream
+git checkout main
+git reset upstream/main --hard
+```
+
+Run `uss-tableflip/scripts/upstream-release <release_version> <old_version>` from cloud-init tree with updated main branch.
+The script will:
+* Prompt you to create the bug you created in the previous section (it doesn't create it for you)
+* Generate the contents to be pasted into the bug description. At this point you should finish creating the bug. You can finish the TODO later
+* Prompt you to enter the bug number of the created bug
+* Create a local release branch containing a release commit and an updated changelog
+
+**Note:** If a branch other than main was used or main is out of date,
+you will get strange entries in Changelog about cherry pick commits or
+commits will not match tip of upstream/main.
+
+**Note:**  At this point, running `tox` locally will fail, complaining
+that your version does not match what git-describe does.  CI is
+configured to work around this problem for release branches, so you
+can disregard these local failures.  (We will create a tag later in
+the process which fixes this issue.)
+
+Push the branch up for review and create a pull request.  We will use that PR for some documentation on things that have been tested.
+
+## Fill in launchpad bug highlights
+Now that your PR is up and ready for review, go back to the launchpad bug you created and fill in the highlights. There's no real formula for this other than a bulleted list of the 5-ish most noteworthy changes this release. Generally this won't include testing or simple bug fixes.
+
+## Merge branch and tag
+After getting approval for your release branch, merge the branch to main, then tag (annotated and signed):
+```bash
+$ git tag --annotate --sign -m 'Release <version>' <version>
+```
 Then push it:
+```bash
+$ git push upstream <version>
+```
+Note that if the reviewer merged your branch, they may have tagged it as well, so check the tags before tagging it yourself.
+
+# Upstream Release
+**WAIT UNTIL THE RELEASE BRANCH HAS BEEN MERGED BEFORE PERFORMING THIS SECTION**
+
+## Create release tarball
+Tarball *must* be signed.
+```bash
+$ ./tools/make-tarball
+cloud-init-<version>.tar.gz
+
+$ gpg --sign --armor --detach-sig cloud-init-<version>.tar.gz
+```
+
+## Create release in Launchpad
+Example of a finished release: https://launchpad.net/cloud-init/trunk/21.3
+### Option 1: Script
+* Copy the changelog (for *just* this release) to a file called 'changelog-file'
+* Copy the release notes (not including changelog) from the launchpad bug to a file called 'releasenotes-file'
+```bash
+$ lp-project-upload cloud-init <version> cloud-init-<version>.tar.gz <NEXT_version> changelog-file releasenotes-file
+```
+Note that `<NEXT_version>` in the command is for specifying our next milestone. So if we're currently releasing 21.3, `<NEXT_version>` would be 21.4.
+### Option 2: Launchpad UI
```

Comment:
Thank you for this separate manual set of steps to help out as lp-project-upload has had issues in the past.